src/main/java/cz/cvut/kbss/termit/service/business/SearchService.java

```diff
@@ -38,9 +43,18 @@ public SearchService(SearchDao searchDao) {
      * Executes full text search in assets.
      *
      * @param searchString String to search by
+     * @param vocabularies URIs of vocabularies to search in, or null, if all vocabularies shall be searched
+     * @param termsOnly whether to return only matches related to terms (not vocabularies)
      * @return Matching assets
      */
-    public List<FullTextSearchResult> fullTextSearch(String searchString) {
-        return searchDao.fullTextSearch(searchString);
+    public List<FullTextSearchResult> fullTextSearch(String searchString, Set<URI> vocabularies, boolean termsOnly) {
```

Comment:
Not a good idea using boolean switches, especially given the quite significantly different nature of the methods. Two methods explicitly specifying the implemented functionality should be used IMHO.

Reply:
Expected uses of the method:
- vocabularies = null, termsOnly = false - implementation of FTS so far
- vocabularies = null, termsOnly = true - we don't have a use-case yet.
- vocabularies != null, termsOnly = false - we don't have a use-case yet.
- vocabularies != null, termsOnly = true - implementation of FTS for IPR

So I agree, will remove the switch and create a separate method.